pennylane/_qubit_device.py

```diff
@@ -71,8 +71,9 @@ class QubitDevice(Device):
         wires (int, Iterable[Number, str]]): Number of subsystems represented by the device,
             or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
             or strings (``['ancilla', 'q1', 'q2']``). Default 1 if not specified.
-        shots (int): number of circuit evaluations/random samples used to estimate
-            expectation values of observables
+        shots (int, list[int]): Number of circuit evaluations/random samples used to estimate
+            expectation values of observables. If a list of integers is passed, the circuit
+            evaluations are batched over the list of shots.
```

Comment:
I think this will need some more explanation? For example, what is the rule for circuit outputs? If the device is analytic and `qml.probs(), qml.sample()` is returned, will only the second return value be batched? But will both be batched if the device is non-analytic?

I know that this will somewhat change by shots refactor II, but good to be precise?

Reply:
Yes, good point. At the moment, anything that relies on samples will get batched if the shots are batched. So that includes samples always, and expval/var/probs in non-analytic mode.

This will definitely be more intuitive after II, because the presence of batched shots will _always_ correspond to batched output.